test/byContract/IexecEscrow/IexecEscrowNative.ts

```diff
@@ -0,0 +1,302 @@
+                    DEPOSIT_AMOUNT_ETHER,
+                );
+            });
+        });
+
+        describe('Deposits', () => {
+            it('Should deposit native tokens', async () => {
+                await checkBalances(anyone.address, BigNumber.from(0));
+                const initialBalance = await ethers.provider.getBalance(anyone.address);
+
+                expect(await iexecPocoAsAnyone.callStatic.deposit({ value: DEPOSIT_AMOUNT_ETHER }))
+                    .to.be.true;
+                await expect(iexecPocoAsAnyone.deposit({ value: DEPOSIT_AMOUNT_ETHER }))
+                    .to.changeEtherBalances(
+                        [anyone, iexecPoco],
+                        [-DEPOSIT_AMOUNT_ETHER, DEPOSIT_AMOUNT_ETHER],
+                    )
+                    .to.emit(iexecPocoAsAnyone, 'Transfer')
+                    .withArgs(ZERO_ADDRESS, anyone.address, DEPOSIT_AMOUNT);
+
+                await checkBalances(anyone.address, DEPOSIT_AMOUNT);
+                expect(await ethers.provider.getBalance(anyone.address)).to.equal(
+                    initialBalance.sub(DEPOSIT_AMOUNT_ETHER),
+                );
+                expect(await ethers.provider.getBalance(iexecPoco.address)).to.equal(
+                    DEPOSIT_AMOUNT_ETHER,
+                );
+            });
+
+            it('Should deposit native tokens for another address', async () => {
+                await checkBalances(iexecAdmin.address, BigNumber.from(0));
+                await checkBalances(anyone.address, BigNumber.from(0));
+
+                expect(
+                    await iexecPoco.callStatic.depositFor(anyone.address, {
+                        value: DEPOSIT_AMOUNT_ETHER,
+                    }),
+                ).to.be.true;
+                await expect(iexecPoco.depositFor(anyone.address, { value: DEPOSIT_AMOUNT_ETHER }))
+                    .to.changeEtherBalances(
+                        [iexecAdmin, iexecPoco],
+                        [-DEPOSIT_AMOUNT_ETHER, DEPOSIT_AMOUNT_ETHER],
+                    )
+                    .to.emit(iexecPocoAsAnyone, 'Transfer')
+                    .withArgs(ZERO_ADDRESS, anyone.address, DEPOSIT_AMOUNT);
+
+                await checkBalances(iexecAdmin.address, BigNumber.from(0));
+                await checkBalances(anyone.address, DEPOSIT_AMOUNT);
+                expect(await ethers.provider.getBalance(iexecPoco.address)).to.equal(
+                    DEPOSIT_AMOUNT_ETHER,
+                );
+            });
+
+            it('Should depositForArray with exact value and good array lengths', async () => {
+                const amounts = [DEPOSIT_AMOUNT.mul(2), DEPOSIT_AMOUNT];
+                const totalAmount = amounts.reduce((a, b) => a.add(b), BigNumber.from(0));
+                const nativeAmountSent = ethers.utils.parseUnits(totalAmount.toString(), 9);
+                const targets = [iexecAdmin.address, anyone.address];
```

Comment:
Maybe it is more natural to test with 3 different accounts A -> B, C instead of A -> A, B, what do you think?

Reply:
Done